### PR TITLE
feat: User sees latest editorial widget GRO-691

### PR DESCRIPTION
**Pull request by jonallured**
This PR is the last of a trio that adds iOS widgets to Eigen. This PR is for the editorial version of the widget. I'm calling it `LatestEditorial` and it joins `FeaturedArtworks` and `FullBleed` as our initial widget offerings for users. Here they all are on my iPad:

<details>

IMG_0037

</details>

## Source of editorial articles

I chose the news editorial feed as the source of latest editorial articles:

https://www.artsy.net/rss/news

This is an XML file that's updated when we publish new content. The approach is to parse this file, take the first 4 items and run them in the widget. Like the featured artworks widget, the array of four items is rotated throughout the day when one has installed the large or medium versions. This keeps the widget looking fresh.

## Parsing XML

Unlike JSON the Swift standard library has no built-in XML parsing features that connect to structs so I had to use the `XMLParser` class:

https://developer.apple.com/documentation/foundation/xmlparser

This class uses the delegate pattern so I wrote an `ArticleParser` class that implements the pattern and then from there the overall process is mostly the same as the JSON one.

https://artsyproduct.atlassian.net/browse/GRO-691

/cc @artsy/grow-devs

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **It seems like you added some non WebP images to Eigen, please convert them to WebPs using `source images/script.sh` script **
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 247e7ad6eb2be8cd50faa05a3d1c782684b7b868
</p>

File: ArtsyWidget/FullBleed/FullBleed+View.swift [hunk 1]
```diff
@@ -14,11 +14,10 @@ extension FullBleed {
         
         var body: some SwiftUI.View {
             let artsyLogo = UIImage(named: "WhiteArtsyLogo")!
-            
             let artworkImage = artwork.image!
             let artistName = artwork.artist.name
             let artworkTitle = artwork.title
-            let artworkUrl = ArtworkUrl.from(slug: artwork.id)
+            let artworkUrl = artwork.url
```
Comment (pvinis) [inline ArtsyWidget/FullBleed/FullBleed+View.swift:20]: nice touch

File: Artsy.xcodeproj/xcshareddata/xcschemes/ArtsyWidgetExtension.xcscheme [hunk 2]
```diff
@@ -57,31 +57,32 @@
...
+      <RemoteRunnable
+         runnableDebuggingMode = "2"
+         BundleIdentifier = "com.apple.springboard">
          <BuildableReference
             BuildableIdentifier = "primary"
-            BlueprintIdentifier = "49BA7DFE1655ABE600C06572"
-            BuildableName = "Artsy.app"
-            BlueprintName = "Artsy"
+            BlueprintIdentifier = "FCB45B212746EF3700232337"
+            BuildableName = "ArtsyWidgetExtension.appex"
```
Comment (pvinis) [inline Artsy.xcodeproj/xcshareddata/xcschemes/ArtsyWidgetExtension.xcscheme:66]: I wonder what happened here 😅

**Review by pvinis — APPROVED** [review]
very nice, but man, woof, the amount of code for these widgets grew exponentially!

File: Artsy.xcodeproj/xcshareddata/xcschemes/ArtsyWidgetExtension.xcscheme (hunk 2 above)
Comment (jonallured) [reply Artsy.xcodeproj/xcshareddata/xcschemes/ArtsyWidgetExtension.xcscheme:66]: Yeah no clue!

**Comment by jonallured** [comment]
Merging this and happy to come back to anything if people end up with feedback they'd like to see addressed!! 🙌

**Pull request merged**